pkg/ccl/backupccl/backup_destination.go

```diff
@@ -353,14 +388,44 @@ func writeNewLatestFile(
 	ctx context.Context, settings *cluster.Settings, exportStore cloud.ExternalStorage, suffix string,
 ) error {
 	// If the cluster is still running on a mixed version, we want to write
-	// to the base directory as well the progress directory. That way if
-	// an old node resumes a backup, it doesn't have to start over.
+	// to the base directory instead of the metadata/latest directory. That
+	// way an old node can still find the LATEST file.
 	if !settings.Version.IsActive(ctx, clusterversion.BackupDoesNotOverwriteLatestAndCheckpoint) {
-		err := cloud.WriteFile(ctx, exportStore, latestFileName, strings.NewReader(suffix))
+		return cloud.WriteFile(ctx, exportStore, latestFileName, strings.NewReader(suffix))
+	}
+
+	// HTTP storage does not support listing and so we cannot rely on the
+	// above-mentioned List method to return us the most recent latest file.
+	// Instead, we will write a latest file once with a well-known filename,
+	// and teach the job to always reach for that filename in the face of
+	// a resume. We may lose progress, but this is a cost we are willing
```

Comment:
nit: comment talks about losing progress but this is the LATEST file that tracks the latest full backup in the collection. Modify comment to reflect that since this is only ever written once anyways.